### PR TITLE
add sharded to querystats bytes processed

### PR DESCRIPTION
**Pull request by trevorwhitney**
This adds a shards label to the `loki_logql_querystats_bytes_processed_per_second` metric that we use for determining our `LokiQueryRangeReadPerformanceBudgetBurn` metric so we can exclude slow queries with 0 shards from burning our SLO budget.

WDYT?

### CODE REVIEW

**Comment by chaudum** [comment]
> WDYT?

What is the potential cardinality of the `shards` label?

**Comment by trevorwhitney** [comment]
@chaudum that's a good point, it's bounded by the number of shards which could be in the thousands. What if we converted this to a boolean, as we really only care about `0 shards` or `> 0 shards`?

**Comment by trevorwhitney** [comment]
I'm thinking `shardable=[true|false]`

File: pkg/logql/metrics.go [hunk 1]
```diff
@@ -260,7 +264,8 @@ func RecordSeriesQueryMetrics(
 		"total_entries", stats.Summary.TotalEntriesReturned,
 	)
 
-	bytesPerSecond.WithLabelValues(status, queryType, "", latencyType).
```
Comment (DylanGuedes) [inline pkg/logql/metrics.go:263]: doesn't that mean that this metric can have infinite cardinality?

**Review by DylanGuedes — CHANGES_REQUESTED** [review]

**Comment by owen-d** [comment]
Make it `sharded=true|false` because shardable queries that touch less than 1 shard's maximum size will be executed unsharded.

**Comment by trevorwhitney** [comment]
I have changed the label to a boolean, with the logic being 0 or 1 shards is `sharded=false`, while > 1 shards is `sharded=true`. There are a few places in the code we check if shards are 0 and if so set them to 1, so it seemed like we'd what to capture both cases for "non-sharded".

File: pkg/logql/metrics.go [hunk 2]
```diff
@@ -133,7 +134,7 @@ func RecordRangeAndInstantQueryMetrics(
 		"store_chunks_download_time", stats.ChunksDownloadTime(),
 		"queue_time", logql_stats.ConvertSecondsToNanoseconds(stats.Summary.QueueTime),
 		"splits", stats.Summary.Splits,
-		"shards", stats.Summary.Shards,
+		"sharded", stats.Summary.Shards,
```
Comment (DylanGuedes) [inline pkg/logql/metrics.go:137]:
```suggestion
		"shards", stats.Summary.Shards,
```

**Review by DylanGuedes — CHANGES_REQUESTED** [review]

File: pkg/logql/metrics.go (hunk 2 above)
Comment (trevorwhitney) [reply pkg/logql/metrics.go]: sloppy rename... thank you!

**Review by DylanGuedes — APPROVED** [review]

**Pull request merged**